Make contributions easier

jankeromnes opened the pull request:
This pull request configures Gitpod for perf.html development, so that anyone can get started easily with a pre-configured environment.

Workspaces are prepared with `yarn install`, then the perf.html server will start automatically with `PERFHTML_HOST="0.0.0.0" yarn start`. Please let me know if that fits your workflow.

(Hint: When prompted, open the running app in a new Browser tab -- the default Preview won't work because your CSP rules forbid cross-domain iframes. We can make this the default action by adding `onOpen: open-browser` to port 4242 in `.gitpod.yml`.)

Here is what it looks like:

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io#https://github.com/jankeromnes/perf.html)

<img width="1440" alt="screenshot 2019-01-23 at 18 56 39" src="https://user-images.githubusercontent.com/599268/51626799-a6e67580-1f40-11e9-8d97-0e5e81e0876b.png">

@julienw Sorry for incrementing your review pile again! :innocent: Please take your time and review this whenever is convenient for you.

julienw commented on .gitpod.yml:
```diff
@@ -0,0 +1,5 @@
+- port: 4242
+tasks:
+- init: yarn install
+  command: PERFHTML_HOST="0.0.0.0" yarn start
```
Do you know if we can have a second terminal opened, given this terminal will stay busy with the dev server?

julienw replied:
I see we can have several `command` lines that each will open a new terminal: is an empty `command` property legal? Otherwise just an `echo` will work :)

Also maybe adding a "echo" before each command to explain what we're doing could be nice.

jankeromnes replied:
Interesting suggestion, thanks. You can already open new terminals from the UI (`Terminal` > `New Terminal`), but we can indeed automatically open a new terminal to allow newcomers to run commands without having to kill the dev server.

julienw commented on README.md:
```diff
@@ -38,6 +38,10 @@ yarn install
+You can also develop perf.html online in a pre-configured development environment:
+
+[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/devtools-html/perf.html)
+
```
can you please add a note in https://github.com/devtools-html/perf.html/blob/master/CONTRIBUTING.md#getting-started-with-development as well?

jankeromnes commented:
Many thanks for the encouraging review! 🙏 And no worries about the delay (as you can see, it took me a while to notice your reply as well).

jankeromnes commented:
@julienw I believe I've addressed all your nits 😄 

- rebased commit
- added note to `CONTRIBUTING.md` (please let me know if the wording should be changed)
- added a new empty terminal with `- openMode: split-right` (i.e. new empty `command` that opens in a split terminal)
- made port 4242 automatically open a new browser tab with `- onOpen: open-browser` (however, this gets blocked by Nightly's pop-up blocker, so you have to manually allow it)

Please let me know what you think of these changes when you have time.

jankeromnes commented:
> Also maybe adding a "echo" before each command to explain what we're doing could be nice.

That's also an interesting idea. Note that the commands are already printed before getting run in a Gitpod terminal, e.g.:

```bash
gitpod /workspace/perf.html $ {
> yarn install
> } && {
> PERFHTML_HOST="0.0.0.0" yarn start
> }
yarn install v1.12.3
$ node bin/pre-install.js
All project requirements are satisfied, moving forward with the installation.
[1/4] Resolving packages...
[2/4] Fetching packages...
```

Do you have suggestions for what the extra `echo` should say? Maybe something like:

```bash
echo "Starting the perf.html web app:" && PERFHTML_HOST="0.0.0.0" yarn start`
```
?

julienw commented:
I think I was considering a `echo` especially for the second terminal :) Maybe:
```
echo perf.html ❤️ gitpod
echo Welcome to this virtual environment, you can type your commands in this terminal.
```

jankeromnes commented:
Rebased again, and added a `printf`, which looks like this (bottom right terminal):

<img width="1552" alt="screenshot 2019-02-11 at 15 32 12" src="https://user-images.githubusercontent.com/599268/52569817-501ede00-2e12-11e9-86d3-ee9a35ea2e19.png">

I'll also remove the `onOpen: open-browser`, since the blocked pop-up seems like a bad experience (better to let users decide whether to open a Preview pane or new Browser tab when the notification shows up).

You can try the latest version here:

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io#https://github.com/jankeromnes/perf.html)

jankeromnes commented:
Rebased; renamed to "Firefox Profiler"; switched to `node 10+`.

@julienw Please take another look when you have time. 😊

julienw commented on .gitpod.yml:
```diff
@@ -0,0 +1,8 @@
+tasks:
+- before: nvm install 10
+  init: yarn install
+  command: PERFHTML_HOST="0.0.0.0" yarn start
```
```suggestion
  command: FX_PROFILER_HOST="0.0.0.0" yarn start
```

jankeromnes commented:
Oops, indeed, thank you for catching these!

julienw commented:
Just merged, thanks a lot Janx!

jankeromnes commented:
So cool! 🎉 Thanks a lot for powering through all these reviews. 😄 Also please let me know if I can improve the contribution experience in any way!
